btclib/bip32/bip32.py

```diff
@@ -309,43 +309,27 @@ def __init__(
             self.assert_valid()
 
 
-def __prv_key_derivation(xkey: _BIP32KeyData, index: int) -> None:
-    xkey.index = index
-    Q_bytes = bytes_from_point(mult(xkey.prv_key_int))
-    xkey.parent_fingerprint = hash160(Q_bytes)[:4]
-    hmac_ = (
-        hmac.new(
-            xkey.chain_code,
-            xkey.key + index.to_bytes(4, byteorder="big", signed=False),
-            "sha512",
-        ).digest()
-        if xkey.is_hardened
-        else hmac.new(
-            xkey.chain_code,
-            Q_bytes + index.to_bytes(4, byteorder="big", signed=False),
-            "sha512",
-        ).digest()
+def __prv_key_derivation(xkey: _BIP32KeyData, index: int, pub_key: bytes) -> None:
+    xb = (
+        xkey.key
+        if index >= 0x80000000
+        else pub_key or bytes_from_point(mult(xkey.prv_key_int))
     )
+    xb += index.to_bytes(4, byteorder="big", signed=False)
+    hmac_ = hmac.new(xkey.chain_code, xb, "sha512").digest()
     xkey.chain_code = hmac_[32:]
     offset = int.from_bytes(hmac_[:32], byteorder="big", signed=False)
     xkey.prv_key_int = (xkey.prv_key_int + offset) % ec.n
     xkey.key = b"\x00" + xkey.prv_key_int.to_bytes(32, byteorder="big", signed=False)
-    xkey.pub_key_point = INF
 
 
 def __pub_key_derivation(xkey: _BIP32KeyData, index: int) -> None:
-    xkey.index = index
-    xkey.parent_fingerprint = hash160(xkey.key)[:4]
-    hmac_ = hmac.new(
-        xkey.chain_code,
-        xkey.key + index.to_bytes(4, byteorder="big", signed=False),
-        "sha512",
-    ).digest()
+    xb = xkey.key + index.to_bytes(4, byteorder="big", signed=False)
+    hmac_ = hmac.new(xkey.chain_code, xb, "sha512").digest()
     xkey.chain_code = hmac_[32:]
     offset = int.from_bytes(hmac_[:32], byteorder="big", signed=False)
     xkey.pub_key_point = ec.add(xkey.pub_key_point, mult(offset))
     xkey.key = bytes_from_point(xkey.pub_key_point)
-    xkey.prv_key_int = 0
```

Comment:
Same

Reply:
it's a fixed value, no need to update it for each derivation level